runtime/interpreter/value.go

```diff
@@ -16585,6 +16589,22 @@ func (v *CompositeValue) IsDestroyed() bool {
 	return v.isDestroyed
 }
 
+func resourceDefaultDestroyEventName(t sema.ContainerType) string {
+	return resourceDefaultDestroyEventPrefix + string(t.ID())
+}
+
+func (v *CompositeValue) defaultDestroyEventConstructors() (constructors []FunctionValue) {
```

Comment:
I was a bit confused at first why the constructor for the resource destroyed event was not just simply stored by the constant name (`ResourceDestroyed`), but instead is suffixed with the type.

Could you please document that this is necessary to support programs where the resource conforms to interfaces, and the interfaces likewise define resource destroyed events, maybe even with a small example? 

Also, do we have tests for the case of multiple events having to be emitted for the destruction of a single resource?

Reply:
Yes! E.g. https://github.com/onflow/cadence/pull/2817/files#diff-ed4abd754a588f51cf2a7e972e435b1c81c5b4153b25fd927f18ab36d6c429fdR2249